### PR TITLE
Issue #102: make UT to run all from folder

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #102: make UT to run all from folder

### CODE REVIEW

**Comment by HuGanghui** [comment]
Please give me some help, I run `mvn clean verify` on my local, it is good, but CI failed.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/newline/expected.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
-Starting audit...
-[ERROR] /Users/hgh/Downloads/my_file/IDEA_workspace/patchfiltergeneratepatch/src/main/java/Checker/RegexpOnFilename/Test3.java:1: File match folder pattern '' and file pattern '\.java$'. [RegexpOnFilename]
-Audit done.
-
+Test3.java:1: File match folder pattern '' and file pattern '\.java$'.
+Test2.java:1: File match folder pattern '' and file pattern '\.java$'.
+Test.java:1: File match folder pattern '' and file pattern '\.java$'.
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/newline/expected.txt:3]: Expected violations before and after do not match. Probably that's why your test fail.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/patchedline/expected.txt [hunk 2]
```diff
@@ -1,5 +1,3 @@
-Starting audit...
-[ERROR] /Users/hgh/Downloads/my_file/IDEA_workspace/patchfiltergeneratepatch/src/main/java/Checker/RegexpOnFilename/Test3.java:1: File match folder pattern '' and file pattern '\.java$'. [RegexpOnFilename]
-[ERROR] /Users/hgh/Downloads/my_file/IDEA_workspace/patchfiltergeneratepatch/src/main/java/Checker/RegexpOnFilename/Test.java:1: File match folder pattern '' and file pattern '\.java$'. [RegexpOnFilename]
-Audit done.
-
+Test3.java:1: File match folder pattern '' and file pattern '\.java$'.
+Test2.java:1: File match folder pattern '' and file pattern '\.java$'.
+Test.java:1: File match folder pattern '' and file pattern '\.java$'.
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/patchedline/expected.txt:3]: Same here. Expected violations before and after do not match. Probably that's why your test fail.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/newline/expected.txt (hunk 1 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/newline/expected.txt:3]: @baratali ...but It is ok on my local.

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/AbstractPatchFilterEvaluationTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+        // process each of the lines
+        try (ByteArrayInputStream inputStream =
+                     new ByteArrayInputStream(stream.toByteArray());
+             LineNumberReader lnr = new LineNumberReader(
+                     new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
+            final String expectedFile = configPath.replaceFirst(
+                    "(default|zero)ContextConfig.xml", "expected.txt");
+            final List<String> expected = Files.readAllLines(Paths.get(getPath(expectedFile)));
+            final List<String> actuals = lnr.lines().limit(expected.size()).sorted()
```
Comment (HuGanghui) [inline src/test/java/com/puppycrawl/tools/checkstyle/filters/AbstractPatchFilterEvaluationTest.java:88]: at the beginning, I remove the sorted(), and CI failed, after I add it, it ok. Would the default sort be different on different machines?

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/newline/expected.txt (hunk 1 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/newline/expected.txt:1]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/patchedline/expected.txt (hunk 2 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/RegexpOnFilename/patchedline/expected.txt:1]: done

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/AbstractPatchFilterEvaluationTest.java (hunk 3 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/filters/AbstractPatchFilterEvaluationTest.java:88]: This sort is not good, it even made a lot of problems in main project.
We should make sort by numbers of  lines and column, to keep order natural.
Please try to fix it in separate PR if you find this inconvenient too

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/filters/AbstractPatchFilterEvaluationTest.java:88]: We can use Google's Truth library to assert. It awesomely show diff in text.

**Pull request merged**